django/db/backends/mysql/compiler.py

```diff
@@ -36,7 +36,12 @@ def as_sql(self):
 
 
 class SQLUpdateCompiler(compiler.SQLUpdateCompiler, SQLCompiler):
-    pass
+    def as_sql(self):
+        query, params = super().as_sql()
+        if self.query.order_by:
+            sql_list = [sql % tuple(params) for expr, (sql, params, is_ref) in self.get_order_by()]
+            query += ' ORDER BY ' + ', '.join(sql_list)
```

Comment:
You want to avoid interpolating params yourself as that could result in SQL injections.

```suggestion
        if self.query.order_by:
            order_by_sql = []
            order_by_params = []
            for _, (sql, params, is_ref) in self.get_order_by():
                order_by_sql.append(sql)
                order_by_params.extend(params)
            query += ' ORDER BY ' + ', '.join(order_by_sql)
            params += tuple(order_by_params)
```

Reply:
Also I think that `is_ref` will need to be handled in a special way. The reference will need to be resolved to inline the expression.

This can be tested by with the following code

```python
updated_count = UniqueNumber.objects.annotate(
    number_inverse=F('number') * -1
).order_by('number_inverse').update(
    number=F('number') + 2,
)
self.assertEqual(updated_count, 2)
```

In this case `number_inverse` will yield `is_ref=True` and `expr` will be an instance of `Ref` if I'm not mistaken.

Reply:
I'm working on the `QuerySet.annotate().order_by().update()` use-case and I can't seem to reliably call `get_order_by()` without getting an exception since the `self.query.annotations` dict is set to empty whenever the `db.models.query.update()` is called.

As a result, I'm receiving this this exception:
`django.core.exceptions.FieldError: Cannot resolve keyword 'number_inverse' into field. Choices are: id, number`

I'm still digging into this, but I wanted to put it out there in case anyone had any ideas.

Reply:
An option would be to completely ignore such reference then. This use case is already limited and obscure to discover and since `order_by` clauses were previously silently dropped it would preserve backward compatibility?

Reply:
I ended up catching FieldError exceptions in case `get_order_by()` could not be resolved.  This seems to satisfy this use-case.